### ISSUE TITLE
duplicate: Make a kubernetes  ingress

### ISSUE THREAD

**Issue by paulgmiller** [microsoft/reverse-proxy#91]
### What should we add or change to make your life better?
Make and ingress controller that watches kubernetes ingress objects and configures yarp to point at service ips

### Why is this important to you?

Lots of msft teams are running kubernetes and would love and option other than nginx or haproxy

**Comment by davidfowl** [comment]
100% agree. We want to do this!

**Comment by irperez** [comment]
I totally agree, we need this!

**Comment by paulgmiller** [comment]
Be nice to have https://github.com/microsoft/reverse-proxy/issues/8 then you could run your ingress reconciler right inside yarp process. I started down this road but wasn't sure if code configured routes were stable enough. [Based on this](https://devblogs.microsoft.com/dotnet/introducing-yarp-preview-1/#:~:text=The%20configure%20method%20defines%20the%20ASP.NET%20pipeline%20for,added%20to%20customize%20the%20handling%20of%20the%20request.)

In the shorter term you could do a python or go sidecar that spits out a config file of host/path -> kubernetes service dns name.

As long as you goto k8s service  the config won't flux that much. However another step in the future would be to watch endpoints as well to bypass the service name and kube-proxy and go right to healthy endpoints.  But in any large deployment with lots of pods that go healthy/unhealthy often  that's going to make a lot of config flux/reloading so at that point code integration gets more appealing.

Just some thoughts.

**Comment by Tratcher** [comment]
@paulgmiller I'm working on #8 now, we should have you unblocked soon.

**Comment by Tratcher** [comment]
We should de-dupe this against https://github.com/microsoft/reverse-proxy/issues/200 which has more content in it.

**Comment by samsp-msft** [comment]
dupe of #200
